apps/api/src/api/services/transactions/stellar/offrampTransaction.ts

```diff
@@ -41,20 +45,37 @@ export async function buildPaymentAndMergeTx({
     throw new Error("Stellar funding secret not defined");
   }
 
-  const expectedSequenceNumber = await getFutureShiftedLedgerSequence(horizonServer, 32);
-
   const fundingAccountKeypair = Keypair.fromSecret(FUNDING_SECRET);
 
   const { memo, memoType, anchorTargetAccount } = paymentData;
   const transactionMemo = memoType === "text" ? Memo.text(memo) : Memo.hash(Buffer.from(memo, "base64"));
 
   const fundingAccount = await horizonServer.loadAccount(fundingAccountKeypair.publicKey());
 
+  // Define timeframes for each presigned transaction
+  const timeWindows = [
+    SEQUENCE_TIME_WINDOWS.FIRST_TX,
+    SEQUENCE_TIME_WINDOWS.SECOND_TX,
+    SEQUENCE_TIME_WINDOWS.THIRD_TX,
+    SEQUENCE_TIME_WINDOWS.FOURTH_TX,
+    SEQUENCE_TIME_WINDOWS.FIFTH_TX
+  ];
+
+  const sequenceNumbers: string[] = [];
+  for (let i = 0; i < NUMBER_OF_PRESIGNED_TXS; i++) {
+    const sequenceNumber = await getFutureShiftedLedgerSequence(horizonServer, 32, timeWindows[i]);
+    sequenceNumbers.push(sequenceNumber);
+  }
+
+  const expectedSequenceNumber = sequenceNumbers[0]; // Use first sequence number as expected
```

Comment:
This will be put onto the `meta` of each Stellar transaction when we store them in the database. I think it would be better to have each transaction stored with the respective sequence number. This makes it easier for us later on to see what transactions can still be used and which can't in case that we manually need to do something here.